fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -308,23 +307,10 @@ protected Response exceptionResponse(OperationOutcome oo, Status status) {
         }
 
         // Single Location to ensure the Operation Outcomes are Encode.forHtml and avoids any injections.
-        Collection<Issue> currentIssues = oo.getIssue();
-        List<Issue> issues = new ArrayList<>();
-        for (Issue current : currentIssues) {
-            if (current.getDiagnostics() != null) {
-                String diagnostics = current.getDiagnostics().getValue();
-                issues.add(
-                    current.toBuilder()
-                       .diagnostics(string(Encode.forHtml(diagnostics)))
-                    .build());
-            } else {
-                issues.add(current);
-            }
-        }
+        EncodingVisitor<OperationOutcome> v = new EncodingVisitor<>(EncodingContext.HTML);
+        oo.accept(v);
         return Response.status(status)
-                .entity(oo.toBuilder()
-                    .issue(issues)
-                    .build())
+                .entity(v.getResult())
```

Comment:
this is where we use the EncodingVisitor to encode the entire OperationOutcome.  but i'm not sure that we actually want to do this because I slightly prefer the more targeted approach that we attempt today (where we only encode content that might contain user-provided input).  John opened https://github.com/IBM/FHIR/issues/2596 for an idea we had to possibly make that a little more explicit / force us to think about whether an error message could contain user-provided content each time we construct an OperationOutcomeIssue.